imageio/imageio-batik/src/main/java/com/twelvemonkeys/imageio/plugins/svg/SVGImageReaderSpi.java

```diff
@@ -134,10 +136,40 @@ else if (buffer[1] == 'D' && buffer[2] == 'O' && buffer[3] == 'C'
                     if (buffer[0] == 's' && buffer[1] == 'v' && buffer[2] == 'g'
                             && (Character.isWhitespace((char) buffer[3]) || buffer[3] == ':')) {
                         // It's SVG, identified by root tag
-                        // TODO: Support svg with prefix + recognize namespace (http://www.w3.org/2000/svg)!
                         return true;
                     }
 
+                    // Read the full tag name (may contain a prefix of any length)
+                    ByteArrayOutputStream nameBuf = new ByteArrayOutputStream();
```

Comment:
As the default buffer size is 32, we can optimize this by allocating enough space up front. 

```suggestion
                    int MAX_TAG_NAME = 256;
                    ByteArrayOutputStream nameBuf = new ByteArrayOutputStream(MAX_TAG_NAME);
```

Or possibly use 1024, and reuse the buffer for the attribute search?

Reply:
done :)

Reply:
woops